depends/funcs.mk

```diff
@@ -39,8 +39,32 @@ define fetch_file
       $(call fetch_file_inner,$(1),$(FALLBACK_DOWNLOAD_PATH),$(3),$(4),$(5))))
 endef
 
+# Shell script to create a source tarball in $(1)_source from local directory
+# $(1)_local_dir instead of downloading remote sources. Tarball is recreated if
+# any paths in the local directory have a newer mtime, and checksum of the
+# tarball is saved to $(1)_fetched and returned as output.
+define fetch_local_dir_sha256
+    if ! [ -f $($(1)_source) ] || [ -n "$$(find $($(1)_local_dir) -newer $($(1)_source) | head -n1)" ]; then \
```

Comment:
nit: ` | head -n1` is unnecessary

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31741#discussion_r1948608064

> nit: ` | head -n1` is unnecessary

I think it's natural and important to have to prevent bad worst-case behavior. Having `head -n1` lets `find` command exit early and not continue looking for newer files after it has already found some, because its stdout will be closed. Also (probably more importantly) it prevents the shell from having to build a up a huge string with a list of new paths.